### PR TITLE
revisions: refactor uplift request form submission code (Bug 1831384)

### PR DESCRIPTION
**Pull request by cgsheeh**

### CODE REVIEW

File: landoui/revisions.py [hunk 1]
```diff
@@ -101,52 +101,46 @@ def uplift():
...
-                        "repository": repository,
-                    },
-                )
+    if not is_user_authenticated():
+        return jsonify(errors=["You must be logged in to request an uplift"]), 401
+
+    if not uplift_request_form.validate():
+        errors = []
+        for field_errors in uplift_request_form.errors.values():
+            errors.extend(field_errors)
```
Comment (zzzeid) [inline landoui/revisions.py:113]: nit
```suggestion
        errors = list(chain(*uplift_request_form.errors.values()))
```

File: landoui/revisions.py [hunk 2]
```diff
@@ -101,52 +101,46 @@ def uplift():
...
+        return jsonify(errors=errors), 400
 
-                # Redirect to the tip revision's URL.
-                # TODO add js for auto-opening the uplift request Phabricator form.
-                tip_differential = response["tip_differential"]["url"]
-                return redirect(tip_differential)
+    try:
+        revision_id = uplift_request_form.revision_id.data
+        repository = uplift_request_form.repository.data
+    except json.JSONDecodeError as exc:
```
Comment (zzzeid) [inline landoui/revisions.py:119]: It's unclear where `JSONDecodeError` can occur here, looks like we are just fetching existing attributes on `uplift_request_form`?

File: landoui/revisions.py [hunk 3]
```diff
@@ -101,52 +101,46 @@ def uplift():
...
 
-                errors.append(e.detail)
-                return_code = e.status_code
+        return jsonify(errors=[e.detail]), e.status_code
 
-    # If we return an error and we don't hit a block with a specific problem, consider
-    # the error a server-side issue.
-    return jsonify(errors=errors), return_code or 500
+    # Redirect to the tip revision's URL.
+    # TODO add js for auto-opening the uplift request Phabricator form.
```
[the remark this answers is not in the document]
Comment (cgsheeh) [reply landoui/revisions.py:133]: https://bugzilla.mozilla.org/show_bug.cgi?id=1810257 tracks this work.

[the remark this answers is not in the document]
Comment (cgsheeh) [reply landoui/revisions.py:133]: Added a link to the bug near the TODO.

**Review by zzzeid — APPROVED** [review]
lgtm

**Pull request merged**